cli/lsp/documents.rs

```diff
@@ -1248,21 +1242,108 @@ impl Documents {
 
     // only refresh the dependencies if the underlying configuration has changed
     if self.resolver_config_hash != new_resolver_config_hash {
-      self.refresh_dependencies();
+      self.refresh_dependencies(root_dirs);
       self.resolver_config_hash = new_resolver_config_hash;
     }
 
     self.dirty = true;
   }
 
-  fn refresh_dependencies(&mut self) {
+  fn refresh_dependencies(&mut self, root_dirs: Vec<Url>) {
     let resolver = self.resolver.as_graph_resolver();
     for doc in self.open_docs.values_mut() {
       if let Some(new_doc) = doc.maybe_with_new_resolver(resolver) {
         *doc = new_doc;
       }
     }
-    self.file_system_docs.lock().refresh_dependencies(resolver);
+
+    // update the file system document
+    let mut fs_docs = self.file_system_docs.lock();
+    let mut not_found_docs =
+      fs_docs.docs.keys().cloned().collect::<HashSet<_>>();
+
+    let mut pending_dirs = VecDeque::new();
+    for dir in root_dirs {
+      if let Ok(path) = dir.to_file_path() {
+        pending_dirs.push_back(path.to_path_buf());
+      }
+    }
+
+    while let Some(dir_path) = pending_dirs.pop_front() {
+      if let Some(dir_name) = dir_path.file_name() {
+        let dir_name = dir_name.to_string_lossy().to_lowercase();
```

Comment:
**nitpick:** Do we actually want to avoid someone's super special NODE_MODULES folder?

Reply:
It affects enough people that it's worthwhile. We do this with deno fmt/lint/test as well (people need to opt in rather than opt out).